js/tweek-local-cache/src/tweek-repository.ts

```diff
@@ -17,34 +20,36 @@ import {
 import Optional from './optional';
 import MemoryStore from './memory-store';
 import {
+  CachedSingleKey,
   Expiration,
   FlatKeys,
-  StoredKey,
   ITweekStore,
   RefreshErrorPolicy,
+  RepositoryKey,
   RepositoryKeyState,
+  StoredKey,
   TweekRepositoryConfig,
-  CachedSingleKey,
-  RepositoryKey,
 } from './types';
 import exponentIntervalFailurePolicy from './exponent-refresh-error-policy';
 import * as StoredKeyUtils from './stored-key-utils';
-
-export const TweekKeySplitJoin = {
-  split: (key: string) => {
-    return key.toLowerCase().split('/');
-  },
-  join: (fragments: string[]) => fragments.join('/'),
-};
+import { TweekKeySplitJoin } from './split-join';
 
 type KeyValues = { [key: string]: any };
 
 export type Unlisten = () => void;
-export type RepositoryListener = (updatedKeys: string[]) => void;
+export type RepositoryListener = (updatedKeys: Set<string>) => void;
 export type Listen = (listen: RepositoryListener) => Unlisten;
 
+const allowedKeyStates = new Set([RepositoryKeyState.requested, RepositoryKeyState.cached, RepositoryKeyState.missing]);
+
+const createDeprecationWarning = (deprecated: keyof TweekRepository, replacement: keyof TweekRepository) =>
```

Comment:
Maybe implement with a ```@deprecated``` method decorator? it should be easy with typescript and cleaner.